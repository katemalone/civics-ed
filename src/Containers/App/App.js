import React, { Component } from 'react';
import { Route } from 'react-router-dom';
import { connect } from 'react-redux'; 
import { addStatesList, isLoading } from '../../Actions';
import { stateNameList } from '../../utils/helpers';
import StatesContainer from '../StatesContainer/StatesContainer';
import StateInfo from '../StateInfo/StateInfo';
import './App.scss';
import PropTypes from 'prop-types';
import Header from '../../Components/Header'


export class App extends Component {

  componentDidMount() {
    this.getAllStatesList()
  
  }
  
  getAllStatesList = () => {
    this.props.addStatesList(stateNameList)
  }

  render(){
    return(
      <main className='App__main'>
        <Route path='/' render={() => <Header  />} />
        <Route exact path='/' render={() => <StatesContainer />} />
        <Route path='/stateInfo' render={() => <StateInfo /> } />
      </main>
    )
  }
}


export const mapDispatchToProps = dispatch => ({
  addStatesList: data => dispatch(addStatesList(data)),
})

export default connect(null, mapDispatchToProps)(App);


App.propTypes = {
  addStatesList: PropTypes.func,
  isLoading: PropTypes.func
}