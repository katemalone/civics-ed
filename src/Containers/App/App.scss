@import url('https://fonts.googleapis.com/css?family=Public+Sans:300,400,500,700,800,900&display=swap');


* {
  margin: 0;
  padding: 0; 
  color: whitesmoke;
  font-family: 'Public Sans', sans-serif;
  background-color: black;
}


.State_Title{
  color: whitesmoke;
  height: 60px;
}

.StateInfo{
  color: whitesmoke;
}

.StatesCard {
 display:flex;
 flex-direction: column;
 justify-content: center;
 height: 100vh;
 width: 100vw;
}

.StatesCard-div{
  display: flex;
  flex-direction: row;
}

.StatesCard_header {
  width: 40%;
  margin: 6rem 3rem;
}

.StatesCard_img-div {
  display: flex;
  width: 25%;
  justify-content: space-between;
  align-items: center;
  height: 600px;
}

.img_img {
  height: 100%;
}

.arrow {
  width: 30%;
}



.StatesCard_button {
  width: 25%; 
  height: 60px;
  background-color: whitesmoke;
  color: black;
  border-radius: 1rem;
}

