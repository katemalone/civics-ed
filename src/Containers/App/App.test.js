import React from 'react';
import { shallow } from 'enzyme';
import { App } from './App';

describe('App', () => {
let wrapper;
let mockProps;

  beforeEach(() => {
    wrapper = shallow(<App />);
    mockProps= {
      
    }
  }) 

  it('should match Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})