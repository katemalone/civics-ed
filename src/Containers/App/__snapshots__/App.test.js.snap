// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapShot 1`] = `
<main
  className="App__main"
>
  <Route
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    path="/stateInfo"
    render={[Function]}
  />
</main>
`;
